x-pack/legacy/plugins/ml/public/data_visualizer/components/field_data_card/field_data_card.tsx

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, Fragment } from 'react';
+import {
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+  EuiLoadingSpinner,
+  EuiSpacer,
+  EuiText,
+} from '@elastic/eui';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { ML_JOB_FIELD_TYPES } from '../../../../common/constants/field_types';
+
+import { FieldVisConfig } from '../../common';
+// @ts-ignore
+import { FieldTitleBar } from '../../../components/field_title_bar';
+import {
+  BooleanContent,
+  DateContent,
+  DocumentCountContent,
+  GeoPointContent,
+  IpContent,
+  KeywordContent,
+  NumberContent,
+  OtherContent,
+  TextContent,
+} from './content_types';
+
+interface Props {
+  config: FieldVisConfig;
+}
+
+export const FieldDataCard: FC<Props> = ({ config }) => {
+  const { fieldName, loading, type, existsInDocs } = config;
+
+  function getCardContent() {
+    switch (type) {
+      case ML_JOB_FIELD_TYPES.NUMBER:
+        if (fieldName !== undefined) {
+          return <NumberContent config={config} />;
+        } else {
+          return <DocumentCountContent config={config} />;
+        }
+
+      case ML_JOB_FIELD_TYPES.BOOLEAN:
+        return <BooleanContent config={config} />;
+
+      case ML_JOB_FIELD_TYPES.DATE:
+        return <DateContent config={config} />;
+
+      case ML_JOB_FIELD_TYPES.GEO_POINT:
+        return <GeoPointContent config={config} />;
+
+      case ML_JOB_FIELD_TYPES.IP:
+        return <IpContent config={config} />;
+
+      case ML_JOB_FIELD_TYPES.KEYWORD:
+        return <KeywordContent config={config} />;
+
+      case ML_JOB_FIELD_TYPES.TEXT:
+        return <TextContent config={config} />;
+
+      default:
+        return <OtherContent config={config} />;
+    }
+  }
+
+  return (
+    <div data-test-subj={`mlFieldDataCard_${fieldName}`}>
+      <div className="mlFieldDataCard">
+        <FieldTitleBar card={config} />
+        <div className="mlFieldDataCard__content">
+          {loading === true && (
+            <Fragment>
```

Comment:
Nit: Loading bit could be broken out into a separate component

Reply:
Good suggestion. I have moved the loading indicator into a separate component, plus have also moved the 'does not exist in docs' content into a separate component in https://github.com/elastic/kibana/pull/42685/commits/31da8259e659eb908cba4922d4ab44bd0dd3898c